src/coreclr/jit/instr.cpp

```diff
@@ -800,6 +800,23 @@ CodeGen::OperandDesc CodeGen::genOperandDesc(GenTree* op)
 
             switch (intrinsicId)
             {
+                case NI_AVX2_BroadcastScalarToVector256:
+                {
+                    assert(hwintrinsic->isContained());
+                    if (hwintrinsic->IsEmbBroadcast())
+                    {
+                        op = hwintrinsic->AsHWIntrinsic()->Op(1)->AsHWIntrinsic()->Op(1);
+                        assert(op->OperIs(GT_LCL_VAR) && op->TypeIs(TYP_FLOAT));
```

Comment:
This assert is going to be too restrictive.

The API shape is that we take a `Vector128<T>` as input and so one of the most common cases will be that `op1` is itself a contained `CreateScalarUnsafe` node.

Reply:
In general we could see and should support other types of operands as well, with the ultimate decision being based on whether or not `op1` comes from memory or not.

Reply:
I am trying to understand more clearly, so the key point we should check here is that if there is a contained `CreateScalarUnsafe` as the broadcast operand?

Reply:
That's one of the patterns we'll need to handle.

I believe you can simplify it down to effectively:
```
if (hwintrinsic->isContained())
{
    // We're contained, which can only happen as part of a node which supports embedded broadcast
    op = hwintrinsic->AsHWIntrinsic()->Op(1);

    // The broadcast's op1 should also be contained, so we know its coming from memory
    assert(op1->isContained());
}
```

Reply:
Indeed, I agree. Thanks for the feedback.